frontend/src/lib/components/LemonDialog/LemonDialog.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react'
+import { LemonButton, LemonButtonProps } from 'lib/components/LemonButton'
+import { LemonModal, LemonModalProps } from '../LemonModal'
+import ReactDOM from 'react-dom'
+
+export type LemonDialogButtonProps = Pick<LemonButtonProps, 'type' | 'status' | 'icon' | 'onClick'> & {
+    children: React.ReactNode
+}
+
+export type LemonDialogProps = Pick<LemonModalProps, 'title' | 'description' | 'width' | 'inline'> & {
+    primaryButton?: LemonDialogButtonProps
+    secondaryButton?: LemonDialogButtonProps
+    tertiaryButton?: LemonDialogButtonProps
+    onClose?: () => void
+    onAfterClose?: () => void
+}
+
+export function LemonDialog({
+    onAfterClose,
+    onClose,
+    primaryButton,
+    tertiaryButton,
+    secondaryButton,
+    ...props
+}: LemonDialogProps): JSX.Element {
+    const [isOpen, setIsOpen] = useState(true)
+
+    primaryButton = primaryButton || {
+        children: 'Okay',
```

Comment:
I'm a bit uncomfortable with how generic the default "Okay" is. 😅 We should be specific with affirmative buttons in most cases.

Reply:
I'm pretty certain it will never get used but it's a fair point 👍